### PR TITLE
#166238072 Users should be able to create their profiles

### PR DESCRIPTION
**Pull request by Nerocodes**
#### What does this PR do?
  - Adds View, Edit and Create user endpoints
  - Adds Tests for this endpoints
  - Updates swagger docs

#### Description of Task to be completed?
  - Users should be able to create their profiles
  > GET api/v1/userProfile/:id
  > PATCH api/v1/userProfile/:id

#### How should this be manually tested?
  > Clone the repo and cd into the directory
  > Run npm install
  > Run npm run Start-dev
  > Use postman to test all endpoints above
  > key: Content-Type value: application/json

#### What are the relevant pivotal tracker stories?
[#166238072](https://www.pivotaltracker.com/story/show/166238072)

##### View User Profile
deferral-view-userProfile

##### Edit User Profile
deferral-edit-userProfile

### CODE REVIEW

**Review by akpante3 — APPROVED** [review]

**Review by chialuka — APPROVED** [review]
Good job @Nerocodes. Well done

**Review by ayodejiAA — APPROVED** [review]
Awesome job @Nerocodes

File: server/controllers/userController.js [hunk 1]
```diff
@@ -106,4 +109,61 @@ export default class UsersController {
...
+  static async viewUserProfile(req, res, next) {
+    try {
+      const { id } = req.params;
+      const userProfile = await viewProfile(id);
+      if (!userProfile) {
+        return badPostRequest(res, 404, {
+          message: 'Profile not found'
+        });
+      }
+      return successfulRequest(res, 200, { ...userProfile });
```
Comment (tolulope-od) [inline server/controllers/userController.js]: Great work here Nero! I like how you added the swagger definition for the UserProfile, however, some of your responses, don't exactly follow the specification as it's missing a `data` field when sending back a response with data. Please refer to [this](https://github.com/omniti-labs/jsend) specification for more guidance. Once again, Nice job!

**Review by tolulope-od — APPROVED** [review]

File: server/controllers/userController.js (hunk 1 above)
Comment (tolulope-od) [reply server/controllers/userController.js]: Forgive me, I just noticed the screenshot had the right responses! Great Job Nero

**Pull request merged**